core/sourceCode.ml

```diff
@@ -86,7 +86,35 @@ module Position = struct
     code : source_code option;
   }
 
-  let pp fmt _ = Utility.format_omission fmt
+
+  let pp : Format.formatter -> t -> unit = fun fmt pos ->
+    let pp_non_dummy () =
+      let file = pos.start.Lexing.pos_fname in
+
+      Format.fprintf fmt "File %s, "  file;
+
+      let start_line  = pos.start.Lexing.pos_lnum  in
+      let start_char  = pos.start.Lexing.pos_cnum  in
+      let finish_line = pos.finish.Lexing.pos_lnum in
+      let finish_char = pos.finish.Lexing.pos_cnum in
+
+      if start_line = finish_line then
```

Comment:
I don't like these kind of nested `if`s. Perhaps it would be better to:
```
match (start_line = finish_line, start_char = finish_char) with
| (true, true ) -> ...
| (true, false) -> ...
| (false, _   ) -> ...
```

Not sure though. This isn't a critical fragment of the code so I guess either way will work.

Reply:
I personally don't mind the nested `if`, but if you really want to rewrite them, then you can avoid an allocation by doing something along the lines of.
```ocaml
match start_line = finish_line with
| true when start_char = finish_char -> ...
| true -> ...
| false -> ...
```

Reply:
I have an aversion to pattern matching on booleans and prefer the nested ifs, but that's just personal preference.

Reply:
I see the point in trying to avoid nested `if`s, but I'd say in this case where the branches are quite short it should be okay.

Reply:
Ok, dismiss my remark then.